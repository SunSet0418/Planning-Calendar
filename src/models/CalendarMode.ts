export enum CalendarMode {
    Month = '월',
    Week = '주'
}