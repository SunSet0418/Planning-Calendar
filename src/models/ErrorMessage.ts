export class ErrorMessage {
    constructor(
        public title: string,
        public description: string
    ) { }
}