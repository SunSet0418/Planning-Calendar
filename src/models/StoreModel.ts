import { TaskStoreModel } from '../modules/reducers/task';

export interface StoreModel {
    task: TaskStoreModel
}