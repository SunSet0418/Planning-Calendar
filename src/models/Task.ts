export interface Task {
    id: number
    startDate: string,
    endDate: string,
    startTime: number,
    durationTime: number
}