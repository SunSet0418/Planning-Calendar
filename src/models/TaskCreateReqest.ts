export interface TaskCreateRequest {
    name: string,
    date: string,
    startHour: number,
    endHour: number
}