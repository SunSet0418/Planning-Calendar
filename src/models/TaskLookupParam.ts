export class TaskLookupParam {
    constructor(
        public year: number,
        public month: number
    ) {}
}