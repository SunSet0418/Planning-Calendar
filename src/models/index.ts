export { TaskStore } from './TaskStore';
export { Task } from './Task';
export { StoreModel } from './StoreModel';
export { CalendarMode } from './CalendarMode';
export { ErrorMessage } from './ErrorMessage';