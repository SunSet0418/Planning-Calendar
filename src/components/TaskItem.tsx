import * as React from 'react';
import styled from 'styled-components';
import { Button, Tooltip } from 'antd';
import 'antd/lib/button/style/css';
import 'antd/lib/tooltip/style/css';
import { Task } from '../models';

const TaskItemContainer = styled(Button)`
    width: 100%;
    height: auto;
    transition: none;
`

interface OwnProps {
    task: Task
}

type Props = OwnProps & React.HTMLAttributes<HTMLDivElement>;

export const TaskItem: React.FC<Props> = ({ task, ...divProps }) => {
    const tooltip = `${task.date} ${task.startHour}시 ~ ${task.endHour}시`;
    return (
        <Tooltip title={tooltip}>
            <TaskItemContainer {...divProps} data-taskid={task.id} draggable={true}>
                {task.name}
            </TaskItemContainer>
        </Tooltip>
    )
}  