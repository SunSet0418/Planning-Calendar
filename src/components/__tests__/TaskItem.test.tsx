import * as React from 'react';
import * as Enzyme from 'enzyme';
import { TaskItem } from '../TaskItem';
import { Task } from '../../models';

describe('<TaskItem />', () => {
    let component = null;
    const task = new Task(1, '이름', '2019-10-11', 2, 3);

    it('renders correctly', () => {
        component = Enzyme.shallow(<TaskItem task={task}/>);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});