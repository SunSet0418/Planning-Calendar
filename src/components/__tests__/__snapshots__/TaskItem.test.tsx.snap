// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskItem /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    task={
        Task {
            "date": "2019-10-11",
            "endHour": 3,
            "id": 1,
            "name": "이름",
            "startHour": 2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "arrowPointAtCenter": false,
      "autoAdjustOverflow": true,
      "children": <Unknown
        data-taskid={1}
        draggable={true}
>
        이름
</Unknown>,
      "mouseEnterDelay": 0.1,
      "mouseLeaveDelay": 0.1,
      "placement": "top",
      "title": "2019-10-11 2시 ~ 3시",
      "transitionName": "zoom-big-fast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "이름",
        "data-taskid": 1,
        "draggable": true,
      },
      "ref": null,
      "rendered": "이름",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Group": [Function],
        "__ANT_BUTTON": true,
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    width: 100%;
    height: auto;
    transition: none;
",
          ],
        },
        "displayName": "Styled(Button)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <Unknown
          data-taskid={1}
          draggable={true}
>
          이름
</Unknown>,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "top",
        "title": "2019-10-11 2시 ~ 3시",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "이름",
          "data-taskid": 1,
          "draggable": true,
        },
        "ref": null,
        "rendered": "이름",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "__ANT_BUTTON": true,
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
    width: 100%;
    height: auto;
    transition: none;
",
            ],
          },
          "displayName": "Styled(Button)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
